core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -457,9 +457,9 @@ private[spark] object HadoopRDD extends Logging {
       if (locationStr != "localhost") {
         if (loc.isInMemory) {
           logDebug(s"Partition $locationStr is cached by Hadoop.")
-          Some(HDFSCacheTaskLocation(locationStr).toString)
+          Option(HDFSCacheTaskLocation(locationStr).toString)
```

Comment:
I don't see how the result of .toString can be null. Isn't the issue if anything that HDFSCacheTaskLocation.apply returns null?

Reply:
Yeah, I think this can't be null.

Reply:
shall we filter cases that locationStr is null first, or leave it `hdfs_cache_null` as a result?

Reply:
Filtering the null cases would be a good approach.